themes/default/content/learn/abstraction-encapsulation/component-resources/index.md

```diff
@@ -0,0 +1,113 @@
+---
+title: "Building a Component Resource"
+layout: topic
+date: 2021-11-17
+draft: false
+description: |
+    Try creating a component resource, or a logical grouping of code that Pulumi
+    recognizes as a resource. 
+meta_desc: |
+    Try creating a component resource, or a logical grouping of code that Pulumi
+    recognizes as a resource. 
+index: 3
+estimated_time: 10
+meta_image: meta.png
+authors:
+    - laura-santamaria
+tags:
+    - learn
+links:
+    - text: Component Resources
+      url: https://www.pulumi.com/docs/intro/concepts/resources/#components
+block_external_search_index: true
+---
+
+Now that we have a better understanding of why abstraction and encapsulation are
+valuable concepts for Pulumi programs, let's explore what happens when we start
+working in larger teams at scale.
+
+A resource in Pulumi terms is basically a building block. That building block
+could be a base component like an `s3.Bucket` object that's managed by external
+providers like your favorite cloud provider. Alternatively, a resource could be
+a group of other resources that implements abstraction and encapsulation for
+reasons like defining a common pattern. This latter type of resource is called a
+_Component Resource_ in Pulumi's terminology.
+
+## Deciding to create a component resource
+
+Why would you make a component resource instead of just a "simple" logical
+grouping that you can call and use? A component resource shows up in the Pulumi
+ecosystem just like any other resource. That means it has a trackable state,
+appears in diffs, and has a name field you can reference that refers to the
+entire grouping.
+
+We've actually already started creating a component resource in the last part
+when we made a new class that built up an `s3.Bucket` and an `s3.BucketPolicy`
+with a `define_policy` function. Let's now turn that logical grouping into an
+actual component resource.
+
+## Converting to a component resource
+
+When we're converting to a component resource, we're subclassing (or composing
+from an anonymous field, if you're into Go) the `ComponentResource` so that our
```

Comment:
I think it's ok to ignore Go and just focus on Python since the implementation is pretty different in each language IMO.